### DISCUSSION TITLE
How to maintain developer and production composer in one file?

### DISCUSSION THREAD

**Discussion by mucholove** [composer/composer#11849] [Q&A / Support]
So I am developing both my project and several libraries at the same time. This means that my libraries have both local versions I am using, and also canonincal versions on the network.

My intention is to be able to leverage this composer file for local development, and also use it such that my deploy pipeline runs "composer install" and we're good. However, fresh installs of a particular app do no usually have local versions of the app available in the folder above.

Would love to know how to best approach this.
 
This is what it looks like in my `composer.json` right now:

        "repositories": [
        {
            "type": "path",
            "url": "../my-data-access-layer"
        },
        {
            "type": "vcs",
            "url": "https://fossil.mydomain.com/my-data-access-layer"
        },
       ],
      "require": {
             "mucholove/my-data-access-layer": "*"
      }

**Comment by Seldaek** [comment]
I think you probably should look at https://github.com/franzliedke/studio it should help setting up dev copies to keep your composer.json clean and prod-only
